tests/ledger.py

```diff
@@ -109,24 +112,43 @@ def __init__(self, filename):
     def __del__(self):
         self._file.close()
 
+    def unpack_flatbuffer(self, data):
+        frame = kv.Frame.Frame.GetRootAsFrame(data, 0)
+        self._flatbuffer_data = []
+        for i in range(frame.ReplicatedLength()):
+            self._flatbuffer_data.append(frame.Replicated(i))
+        for i in range(frame.DerivedLength()):
+            self._flatbuffer_data.append(frame.Derived(i))
+
+    def _byte_read_fb(self, num_of_bytes):
+        ret = bytes(self._flatbuffer_data[:num_of_bytes])
+        if len(ret) != num_of_bytes:
+            raise ValueError(
+                "Failed to read precise number of bytes: %u" % num_of_bytes
+            )
+        self._flatbuffer_data = self._flatbuffer_data[num_of_bytes:]
```

Comment:
Putting both derived and replicated data in the same buffer is compatible with the current tests now, but probably not what we want to do long term. ledger.py should be usable by someone trying to parse the ledger, and they ought to be able to differentiate replicated from derived data. So should the tests.

Reply:
ok

Reply:
separated into two separate buffers for use if needed